### PR TITLE
Script: fields API for x-pack version, doc version, seq no, mumur3

### PR DESCRIPTION
**Pull request by stu-elastic**
Adds scripting fields API support the rest of the long fields:
* [`_version`](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#index-versioning) - `VersionDocValuesField`
* [`_seq_no`](https://www.elastic.co/guide/en/elasticsearch/reference/master/optimistic-concurrency-control.html) - `SeqNoDocValuesField`
* [`murmur3`](https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-murmur3-usage.html) - `Murmur3DocValueField`
  * Added Painless support to the murmur3 mapper plugin.

All `SortedNumericDocValues` that are interpreted as longs are now subclasses of `AbstractLongDocValuesField`, including murmur, doc version and seq no above as well as `LongDocValuesField` and `UnsignedLongDocValuesField`

Also adds:
* [x-pack's version](https://www.elastic.co/guide/en/elasticsearch/reference/master/version.html) - `VersionStringDocValuesField`
  * Created new `Version` value type as a location for future helpers for comparing versions. 
    * Implements `toString` for the expected representation of the version
  * Implements `asString(String)` and `asString(int, String)`, `asStrings()` converters on field.

Refs: #79105

### CODE REVIEW

**Comment by stu-elastic** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

File: plugins/mapper-murmur3/src/main/java/org/elasticsearch/script/field/murmur3/Murmur3PainlessExtension.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    private static final Whitelist WHITELIST = WhitelistLoader.loadFromResourceFiles(
+        Murmur3PainlessExtension.class,
+        "org.elasticsearch.field.murmur3.txt"
+    );
+
+    @Override
+    public Map<ScriptContext<?>, List<Whitelist>> getContextWhitelists() {
+        List<Whitelist> whitelist = singletonList(WHITELIST);
+        return Map.of(
```
Comment (jdconrad) [inline plugins/mapper-murmur3/src/main/java/org/elasticsearch/script/field/murmur3/Murmur3PainlessExtension.java:38]: Do we know what the behavior of murmur3 is in contexts where it was previously available as a long, but not available as one of the contexts here?

File: server/src/main/java/org/elasticsearch/script/field/IPAddress.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public String toString() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        return builder.value(this.toString());
+        return builder.value(toString());
```
Comment (jdconrad) [inline server/src/main/java/org/elasticsearch/script/field/IPAddress.java:49]: Was this changed intentionally?

File: x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/Version.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+package org.elasticsearch.xpack.versionfield;
+
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Script value class.
+ * TODO(stu): implement {@code Comparable<Version>} based on {@code VersionEncoder#prefixDigitGroupsWithLength(String, BytesRefBuilder)}
```
Comment (jdconrad) [inline x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/Version.java:17]: Please address this TODO.

**Review by jdconrad — APPROVED** [review]
LGTM! Left a few minor comments.

File: x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/Version.java (hunk 3 above)
Comment (stu-elastic) [reply x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/Version.java:17]: Added link to https://github.com/elastic/elasticsearch/issues/82287 in TODO

File: server/src/main/java/org/elasticsearch/script/field/IPAddress.java (hunk 2 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/field/IPAddress.java]: Reverted.

File: plugins/mapper-murmur3/src/main/java/org/elasticsearch/script/field/murmur3/Murmur3PainlessExtension.java (hunk 1 above)
Comment (stu-elastic) [reply plugins/mapper-murmur3/src/main/java/org/elasticsearch/script/field/murmur3/Murmur3PainlessExtension.java]: Added all core contexts and added https://github.com/elastic/elasticsearch/issues/82289

**Pull request merged**